Kernel/Storage/SD/SDMemoryCard.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2023, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Function.h>
+#include <AK/Result.h>
+#include <AK/Types.h>
+#include <Kernel/Locking/Mutex.h>
+#include <Kernel/Storage/SD/Registers.h>
+#include <Kernel/Storage/StorageDevice.h>
+
+namespace Kernel {
+
+class SDHostController;
+
+class SDMemoryCard : public StorageDevice {
+    friend class SDHostController;
+
+public:
+    // ^StorageDevice
+    virtual CommandSet command_set() const override { return CommandSet::SD; }
+
+    // ^BlockDevice
+    virtual void start_request(AsyncBlockDeviceRequest&) override;
+
+private:
+    SDMemoryCard(SDHostController& sdhc, StorageDevice::LUNAddress, u32 hardware_relative_controller_id, u32 block_len, u64 capacity_in_blocks, u32 relative_card_address, SD::OperatingConditionRegister ocr, SD::CardIdentificationRegister cid, SD::SDConfigurationRegister scr);
+
+    enum class CardAddressingMode { ByteAddressing,
+        BlockAddressing };
```

Comment:
Formatting seems a bit weird here, I would have expected something like:
```c++
enum class CardAddressingMode {
    ByteAddressing,
    BlockAddressing,
};
```
This seems to be how we normally format these enums.